beacon_node/eth2_libp2p/src/discovery/subnet_predicate.rs

```diff
@@ -23,10 +24,28 @@ where
                 }
             };
 
-            return bitfield.get(*subnet_id as usize).unwrap_or_else(|_| {
-                                   debug!(log_clone, "Peer found but not on desired subnet"; "peer_id" => format!("{}", enr.peer_id()));
-                false
-            });
+            let matches: Vec<SubnetId> = subnet_ids
+                .clone()
```

Comment:
> why the clone().into_iter() and not just an iter() ?

Updated to just use `iter()`, the borrow checker must have been yelling at me before I updated something else. 

> Could probably set the arg of the function to be a slice also, i.e subnet_ids: &[SubnetId]

I'm having trouble using a slice for the `subnet_ids` arg.  I think it's because the lifetime of the `subnet_predicate` is static.  The error I was getting was "borrow can't outlive anonymous lifetime".